sys/hashes/sha256.c

```diff
@@ -318,3 +318,71 @@ const unsigned char *hmac_sha256(const unsigned char *key,
 
     return result;
 }
+
+unsigned char *sha256_chain(const unsigned char *seed, size_t seed_length,
+                            size_t elements,
+                            unsigned char *tail_element)
+{
+    unsigned char tmp1[SHA256_DIGEST_LENGTH];
+    unsigned char tmp2[SHA256_DIGEST_LENGTH];
+
+    /* if no sha256-chain can be created return NULL */
+    if (elements < 2) {
+        return NULL;
+    }
+
+    /* 1st iteration */
+    sha256(seed, seed_length, tmp2);
+    
+    for (size_t i = 1; i < elements-1; ++i) {
+        if ( (i % 2) == 0 ) {
+            sha256(tmp1, SHA256_DIGEST_LENGTH, tmp2);
+        }
+        else {
+            sha256(tmp2, SHA256_DIGEST_LENGTH, tmp1);
+        }
```

Comment:
` - 2` then and check after the loop. Could still be better than having a condition for every loop run.